From a5b951654b856d899e8c6a4116b4ccc4f18aac91 Mon Sep 17 00:00:00 2001
From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Sun, 24 Feb 2013 17:22:29 -0500
Subject: [PATCH] Add the GNOME Flashback session

This was removed from gnome-session 3.7.4 so let's it add it back here
(under new name)

https://bugzilla.gnome.org/show_bug.cgi?id=694625
---
 data/Makefile.am                        |   18 +++++++++++++++++-
 data/gnome-flashback.desktop.in         |    7 +++++++
 data/gnome-flashback.session.desktop.in |    3 +++
 po/POTFILES.in                          |    2 ++
 4 files changed, 29 insertions(+), 1 deletion(-)
 create mode 100644 data/gnome-flashback.desktop.in
 create mode 100644 data/gnome-flashback.session.desktop.in

diff --git a/data/Makefile.am b/data/Makefile.am
index 83f82e2..84e1954 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -30,17 +30,33 @@ gsettings_SCHEMAS = $(gsettings_schemas_in:.xml.in=.xml)
 
 @GSETTINGS_RULES@
 
+sessiondir = $(datadir)/gnome-session/sessions
+session_in_files = gnome-flashback.session.desktop.in
+session_DATA = $(session_in_files:.session.desktop.in=.session)
+
+xsessiondir = $(datadir)/xsessions
+xsession_in_files = gnome-flashback.desktop.in
+xsession_DATA = $(xsession_in_files:.desktop.in=.desktop)
+
+%.session: %.session.desktop.in   $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; LC_ALL=C $(INTLTOOL_MERGE) -d -u -c $(top_builddir)/po/.intltool-merge-cache $(top_srcdir)/po $< $@
+
+@INTLTOOL_DESKTOP_RULE@
+
 default_layoutdir   = $(datadir)/gnome-panel
 default_layout_DATA = panel-default-layout.layout
 
 EXTRA_DIST =				\
 	$(gsettings_schemas_in_in)	\
+	$(session_in_files)		\
+	$(xsession_in_files)		\
 	$(default_layout_DATA)
 
 CLEANFILES =			\
 	$(gsettings__enum_file)	\
 	$(gsettings_SCHEMAS_in)	\
 	$(gsettings_SCHEMAS)	\
-	*.gschema.valid
+	*.gschema.valid		\
+	$(session_DATA)		\
+	$(xsession_DATA)
 
 -include $(top_srcdir)/git.mk
diff --git a/data/gnome-flashback.desktop.in b/data/gnome-flashback.desktop.in
new file mode 100644
index 0000000..4856f75
--- /dev/null
+++ b/data/gnome-flashback.desktop.in
@@ -0,0 +1,7 @@
+[Desktop Entry]
+_Name=GNOME Flashback
+_Comment=This session logs you into GNOME with the traditional panel
+Exec=gnome-session --session=gnome-flashback
+TryExec=gnome-session
+Icon=
+Type=Application
diff --git a/data/gnome-flashback.session.desktop.in b/data/gnome-flashback.session.desktop.in
new file mode 100644
index 0000000..c012c58
--- /dev/null
+++ b/data/gnome-flashback.session.desktop.in
@@ -0,0 +1,3 @@
+[GNOME Session]
+_Name=GNOME Flashback
+RequiredComponents=gnome-panel;gnome-settings-daemon;gnome-screensaver;metacity;nautilus-classic;notification-daemon;gnome-fallback-background-helper;gnome-fallback-media-keys-helper;gnome-fallback-mount-helper;
diff --git a/po/POTFILES.in b/po/POTFILES.in
index bc65c4c..a6aa6c7 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -23,6 +23,8 @@ applets/wncklet/window-menu.c
 applets/wncklet/wncklet.c
 applets/wncklet/workspace-switcher.c
 [type: gettext/glade]applets/wncklet/workspace-switcher.ui
+data/gnome-flashback.desktop.in
+data/gnome-flashback.session.desktop.in
 data/org.gnome.gnome-panel.applet.clock.gschema.xml.in.in
 data/org.gnome.gnome-panel.applet.fish.gschema.xml.in.in
 data/org.gnome.gnome-panel.applet.window-list.gschema.xml.in.in
-- 
1.7.10.4

